core/src/test/java/fr/sncf/osrd/standalone_sim/ConflictDetectionTest.java

```diff
@@ -250,14 +252,185 @@ In practice, the stop train will require the route in front of him (zone and swi
             var simResultAWithStop =
                     simpleSim(fullInfra, pathPropsA, chunkPathA, 0, Double.POSITIVE_INFINITY, List.of(stop));
 
-            var conflicts = ConflictsKt.detectConflicts(List.of(
-                    convertRequirements(0L, 0.0, simResultAWithStop.train),
-                    convertRequirements(1L, 0.0, simResultB.train)));
+            var reqA = convertRequirements(0L, 0.0, simResultAWithStop.train);
+            var reqB = convertRequirements(1L, 0.0, simResultB.train);
+            var conflicts = ConflictsKt.detectConflicts(List.of(reqA, reqB));
             assertFalse(conflicts.stream().anyMatch((conflict) -> conflict.conflictType == ROUTING));
             assertFalse(conflicts.stream().anyMatch((conflict) -> conflict.conflictType == SPACING));
         }
     }
 
+    /*
+    Test for train starting after the first signal protecting block with potential conflict.
+    Then the train stops either (all combinations):
+    * on stop-signal or not
```

Comment:
Asked myself if it was "stop or not" vs "stop-signal or open signal". After reading again, the structure of the sentence is correct to understand without a doubt, but I would rather make it more explicit anyway.
```suggestion
    * on stop-signal or open signal
```

Reply:
Done in https://github.com/OpenRailAssociation/osrd/pull/7704/commits/66ce90c73d8856e46d02fb8522da75d4a0b5bb54